application/lib/phantoms.go

```diff
@@ -0,0 +1,79 @@
+package lib
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	toml "github.com/pelletier/go-toml"
+)
+
+// ConjurePhantomSubnet - Weighted option to choose phantom address from.
+type ConjurePhantomSubnet struct {
+	Weight  float32
+	Subnets []string
+}
+
+// SubnetConfig - Configuration of subnets for Conjure to choose a Phantom out of.
+type SubnetConfig struct {
+	WeightedSubnets []ConjurePhantomSubnet
+}
+
+// PhantomIPSelector - Object for tracking current generation to SubnetConfig Mapping.
+type PhantomIPSelector struct {
+	Networks map[uint]*SubnetConfig
+}
+
+// type shim because github.com/pelletier/go-toml doesn't allow for integer value keys to maps so
+// we have to parse them ourselves. :(
+type phantomIPSelectorInternal struct {
+	Networks map[string]*SubnetConfig
+}
+
+// var defaultSubnetGeneration = SubnetConfig{
+// 	WeightedSubnets: []ConjurePhantomSubnet{
+// 		{Weight: 9, Subnets: []string{"192.122.190.0/24", "2001:48a8:687f:1::/64"}},
+// 		{Weight: 1, Subnets: []string{"141.219.0.0/16", "35.8.0.0/16"}},
+// 	},
+// }
+
+// func defaultSubnets() *SubnetConfig {
+// 	return &defaultSubnetGeneration
+// }
+
+// GetPhantomSubnetSelector gets the location of the configuration file from an
+// environment variable and returns the parsed configuration.
+func GetPhantomSubnetSelector() (*PhantomIPSelector, error) {
+	return SubnetsFromTomlFile(os.Getenv("PHANTOM_SUBNET_LOCATION"))
```

Comment:
It may help to namespace things and re-evaluate what really deserves to be an env var and what can go in a config file. For the most part they are currently separated by utility.